main/lsp/requests/code_action.cc

```diff
@@ -19,8 +19,12 @@ LSPResult LSPLoop::handleTextDocumentCodeAction(unique_ptr<core::GlobalState> gs
     prodCategoryCounterInc("lsp.messages.processed", "textDocument.codeAction");
 
     core::FileRef file = uri2FileRef(params.textDocument->uri);
-    vector<core::FileRef> files = {file};
-    auto run = tryFastPath(move(gs), {}, files);
+    vector<shared_ptr<core::File>> files;
+    // TODO(sushain): is there another way to do this? maybe a helper I'm missing?
+    files.push_back(make_shared<core::File>(string(file.data(*gs).path()), string(file.data(*gs).source()),
+                                            core::File::Type::Normal));
+    // TODO(sushain): why does tryFastPath(move(gs), {}, files) not work (when files is refs)?
```

Comment:
This should work, and is what you should actually use! What does it do instead?

Reply:
Note that the refs should be the refs received by `uri2FileRef` (e.g., refs active in `GlobalState`).

Reply:
You should just be able to do `tryFastPath(move(gs), {}, {file})` since file refs are copyable.

Reply:
That's exactly what I used initially since it made more sense. However, the tests that I added in the last commit fail that way since `tryFastPath` returns no errors. I suspect it might be tied to duplicate errors since those are also duplicates.